service/http/dffml_service_http/routes.py

```diff
@@ -131,10 +131,35 @@ async def get_mctx(self, request):
 
 
 class HTTPChannelConfig(NamedTuple):
+    """
+    Config for channels.
+
+    Parameters
+    ++++++++++
+        path : str
+            Route in server.
+        dataflow : DataFlow
+            Flow to which inputs from request to path is forwarded too.
+        input_mode : str
+            Mode according to which input data is passed to the dataflow,default:"default".
+                default : Inputs are expected to be mapping of context to list of input
+                        to definition mappings
+                        eg:'{
+                        "insecure-package":
+                            [
+                                {
+                                    "value":"insecure-package",
+                                    "definition":"package"
+                                }
+                            ]
+                        }'
+                "def:NAME" : Input as whole is treated as value for defintion 'NAME'
+    """
```

Comment:
@pdxjohnny is this okay,im not sure if i should have the examples there,also what should i put for `asynchronous` and `presentation`

Reply:
We don't have great documentation for this part of the HTTP server. We should probably create a whole page at `service/http/docs/dataflow.rst` (and include it in the `index.rst`). This documentation should probably just go there (the new .rst) for now (since this docstring won't end up in anything rendered in the docs site at the moment).

For `asynchronous` you can just say that it's unused right now but will be accessible over a websocket in the future and used for long running flows.

For `presentation` you can say that it's how the output will be displayed.